src/gui/socketapi/socketuploadjob.h

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (C) by Hannah von Reth <hannah.vonreth@owncloud.com>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License
+ * for more details.
+ */
+
+#pragma once
+#include <QObject>
+#include <QTemporaryFile>
+
+#include "socketapi.h"
+#include "account.h"
+
+namespace OCC {
+
+class SyncJournalDb;
+class SyncEngine;
+
+class SocketUploadJob : public QObject
+{
+    Q_OBJECT
+public:
+    SocketUploadJob(OCC::SocketListener *listener, const QString &argument, QObject *parent = nullptr);
+
+    void start();
+
+    void finish(const QString &error);
+
+private:
+    SocketListener *_listener;
+    QString _localPath;
+    QString _remotePath;
+    QString _pattern;
+    QTemporaryFile tmp;
+    SyncJournalDb *db;
+    SyncEngine *engine;
+    QStringList syncedFiles;
```

Comment:
Why do the last 4 ones lack the `_` prefix? Confused me a lot during review because I was wondering where these came from.

Reply:
https://code.woboq.org/qt5/qtbase/src/corelib/text/qstring.h.html#QLatin1Literal